packages/@aws-cdk/aws-apigatewayv2-integrations/lib/http/alb.ts

```diff
@@ -22,9 +23,17 @@ export class HttpAlbIntegration extends HttpPrivateIntegration {
   }
 
   public bind(options: HttpRouteIntegrationBindOptions): HttpRouteIntegrationConfig {
+    let vpc: ec2.IVpc | undefined = this.props.vpcLink?.vpc;
+    if (!vpc && (this.props.listener instanceof elbv2.ApplicationListener)) {
+      vpc = this.props.listener.loadBalancer.vpc;
+    }
+    if (!vpc) {
+      throw new Error('The vpcLink property must be specified.');
```

Comment:
I believe we are doing something similar in `_configureVpcLink` helper as well. Do you think we can modify it a little bit to address this use-case (of throwing custom error message from it) ?

Reply:
It's fine to have two different checks.